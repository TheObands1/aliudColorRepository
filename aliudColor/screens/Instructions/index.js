import React, { useState, useEffect}  from 'react';
import { View, Text, Image, TouchableOpacity } from "react-native";
import { GameTitle } from '../../components';
import InstructionsStyle from './styles';

export default function InstructionsView({navigation}){

    function GoBackToMainMenu()
    {
        navigation.navigate("MainMenu");
    }

    return (<View style = {InstructionsStyle.generalStyle}>
                <View style = {{flex: 1, justifyContent: "center",alignItems: "center",}}>
                    <GameTitle fontSize={30}/>
                </View>
                <View style = {{flex: 4}}>
                    <Text style = {InstructionsStyle.instructionsText}>
                    Hey there! In this game you will be presented with a grid who's tiles all have the same color, except
                    for one. You will need to touch the tile that has a different color as fast as you before time runs out.
                    Doing so gives you points, but failing substracts more time out of the timer. Good Luck!
                    </Text>

                    <Text style = {InstructionsStyle.instructionsText}>
                        Note: There's an accesibility button in the main menu to make the game more slower, so that people with
                        voice over on have time to listen to what the voice over tells them and to make the corresponding gestures in order to tap the tiles. 
                    </Text>
                </View>
                <View style = {{flex: 1}}>
                    <TouchableOpacity style={{ alignItems: 'center', marginTop: 10 }} onPress={GoBackToMainMenu} 
                        accessibilityLabel="Go Home Button"
                        accessibilityHint="Takes you back to the main menu">
                        <Image source={require("../../assets/icons/home.png")} style={InstructionsStyle.HomeIcon} />
                    </TouchableOpacity>
                </View>
            </View>);
}