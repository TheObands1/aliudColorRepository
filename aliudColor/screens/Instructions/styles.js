import { StyleSheet } from "react-native";

export default StyleSheet.create({
 generalStyle: {
   flex: 1,
   backgroundColor: "#0a0a0a",
   justifyContent: "center",
   alignItems: "center",
 },
 instructionsText: {
    fontSize: 26,
    color: "#ecf0f1",
    fontFamily: "dogbyte",
    textAlign: 'justify',
    margin: 10
 },
 HomeIcon: {
    width: 50, 
    height: 50
  }
})