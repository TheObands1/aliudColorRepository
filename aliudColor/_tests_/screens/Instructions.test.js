import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import renderer from 'react-test-renderer';

import Instructions from '../../screens/Instructions/index'

describe('tests for the <Instructions/> Screen', () => {

    it('Render the component correctly ', () => {
        const tree = renderer.create(<Instructions />).toJSON();
        expect(tree).toMatchSnapshot();
      });

  it('Navigates to main menu when button is pressed', () => {
    const navigate = jest.fn();
    const { getByAccessibilityLabel } = render(<Instructions navigation={{ navigate }} />);
    fireEvent.press(getByAccessibilityLabel("Go Home Button"));
    expect(navigate).toHaveBeenCalledWith('MainMenu');
  });
});

