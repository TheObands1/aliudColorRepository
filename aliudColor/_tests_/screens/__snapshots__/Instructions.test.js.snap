// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for the <Instructions/> Screen Render the component correctly  1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0a0a0a",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityLabel="aliudColor"
      accessible={true}
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "dogbyte",
            },
            Object {
              "color": "#ecf0f1",
              "fontSize": 30,
            },
          ]
        }
      >
        aliud
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "dogbyte",
            },
            Object {
              "color": "#E64C3C",
              "fontSize": 30,
              "textTransform": "uppercase",
            },
          ]
        }
      >
        c
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "dogbyte",
            },
            Object {
              "color": "#E57E31",
              "fontSize": 30,
            },
          ]
        }
      >
        o
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "dogbyte",
            },
            Object {
              "color": "#F1C431",
              "fontSize": 30,
            },
          ]
        }
      >
        l
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "dogbyte",
            },
            Object {
              "color": "#68CC73",
              "fontSize": 30,
            },
          ]
        }
      >
        o
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "dogbyte",
            },
            Object {
              "color": "#3998DB",
              "fontSize": 30,
            },
          ]
        }
      >
        r
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 4,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ecf0f1",
          "fontFamily": "dogbyte",
          "fontSize": 26,
          "margin": 10,
          "textAlign": "justify",
        }
      }
    >
      Hey there! In this game you will be presented with a grid who's tiles all have the same color, except for one. You will need to touch the tile that has a different color as fast as you before time runs out. Doing so gives you points, but failing substracts more time out of the timer. Good Luck!
    </Text>
    <Text
      style={
        Object {
          "color": "#ecf0f1",
          "fontFamily": "dogbyte",
          "fontSize": 26,
          "margin": 10,
          "textAlign": "justify",
        }
      }
    >
      Note: There's an accesibility button in the main menu to make the game more slower, so that people using text-to-speech have time to listen to what the voice over tells them and to make the corresponding gestures in order to tap the tiles.
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityHint="Takes you back to the main menu"
      accessibilityLabel="Go Home Button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "marginTop": 10,
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 50,
            "width": 50,
          }
        }
      />
    </View>
  </View>
</View>
`;
